Failures with Matplotlib nightly

Recent changes to Matplotlib's `MaxNLocator` (matplotlib/matplotlib#31788) cause Cartopy's `LongitudeLocator` to fail with a TypeError from an unexpected keyword argument.  

We saw this on our testing for a downstream package (log below). 
<details>
  <summary>log</summary>

```
=================================== FAILURES ===================================
__________________________ test_add_lat_lon_gridlines __________________________

args = (), kwargs = {}

    def wrapper(*args, **kwargs):
>       store.return_value[test_name] = obj(*args, **kwargs)
                                        ^^^^^^^^^^^^^^^^^^^^

../../../micromamba/envs/geocat_viz_build/lib/python3.12/site-packages/pytest_mpl/plugin.py:126: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_util.py:52: in test_add_lat_lon_gridlines
    add_lat_lon_gridlines(
../../../micromamba/envs/geocat_viz_build/lib/python3.12/site-packages/geocat/viz/util.py:150: in add_lat_lon_gridlines
    gl = ax.gridlines(
../../../micromamba/envs/geocat_viz_build/lib/python3.12/site-packages/cartopy/mpl/geoaxes.py:1497: in gridlines
    gl = Gridliner(
../../../micromamba/envs/geocat_viz_build/lib/python3.12/site-packages/cartopy/mpl/gridliner.py:255: in __init__
    self.xlocator = LongitudeLocator(dms=dms)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <cartopy.mpl.ticker.LongitudeLocator object at 0x7f9895291010>, nbins = 8
dms = False, kwargs = {}

    def __init__(self, nbins=8, *, dms=False, **kwargs):
>       super().__init__(nbins=nbins, dms=dms, **kwargs)
E       TypeError: MaxNLocator.__init__() got an unexpected keyword argument 'dms'

../../../micromamba/envs/geocat_viz_build/lib/python3.12/site-packages/cartopy/mpl/ticker.py:516: TypeError
=========================== short test summary info ============================
FAILED tests/test_util.py::test_add_lat_lon_gridlines - TypeError: MaxNLocator.__init__() got an unexpected keyword argument 'dms'
========================= 1 failed, 22 passed in 7.39s =========================
Error: Process completed with exit code 1.
```
</details>

I saw similar errors and test failures in the Cartopy test suite when running locally w/ the nightly build of Matplotlib in addition to the image test failures already noted.

Updating the constructor seems to address the issue.  Something like:  
```
    def __init__(self, nbins=8, *, dms=False, **kwargs):
        super().__init__(nbins=nbins, **kwargs)
        self._dms = dms
```

I suspect you'd have caught this sooner rather than later, but figured I would open an issue since I didn't see anything and offer to put in a quick PR.

--- a/lib/cartopy/mpl/ticker.py
+++ b/lib/cartopy/mpl/ticker.py
@@ -513,7 +513,8 @@ class LongitudeLocator(MaxNLocator):
     """
 
     def __init__(self, nbins=8, *, dms=False, **kwargs):
-        super().__init__(nbins=nbins, dms=dms, **kwargs)
+        super().__init__(nbins=nbins, **kwargs)
+        self._dms = dms
 
     def set_params(self, **kwargs):
         """Set parameters within this locator."""

